deploy/003_deploy_swap.ts

```diff
@@ -28,25 +30,30 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   const swapInstance = factorySwapRBTC.attach(deploySwap.address);
   const encodedSwapInitialize = swapInstance.interface.encodeFunctionData("initialize", swapInitializeParams);
 
-  // await swapInstance.callStatic.initialize(swapInitializeParams);
-  // const swapInterface = new ethers.utils.Interface(deploySwap.abi);
-  // swapInterface.encodeFunctionData
+  let constructorArguments = [
+    deploySwap.address,
+    proxyAdminAddress,
+    encodedSwapInitialize
+  ];
 
-  // using ethers
-  // const deployedContract = await upgrades.deployProxy(factorySwapRBTC, [sideWrappedBtcAddress], {
-  //   kind: "transparent",
-  // });
-
-  await deploy('SwapRbtcProxy', {
+  const deployProxyResult = await deploy('SwapRbtcProxy', {
     from: deployer,
     contract: 'TransparentUpgradeableProxy',
-    args: [
-      deploySwap.address,
-      proxyAdminAddress,
-      encodedSwapInitialize
-    ],
+    args: constructorArguments,
     log: true
   });
+
+  if (deployProxyResult.newlyDeployed) {
+    log(`Contract ${SwapRbtcProxy} deployed at ${deployProxyResult.address} using ${deployProxyResult.receipt?.gasUsed.toString()} gas`);
+
+    if(network.live && !isRSK(network)) {
```

Comment:
we are not going to deploy this in a network that is not rsk, so it's not necesary.
In case we where we would need the ETHERSCAN_API_KEY

Reply:
fixed.